backend/src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -182,19 +183,24 @@
                                     bundleSettingsValue.displayName = bundle.getDisplayName();
                                     settingsValues.bundles.put(bundle.getName(), bundleSettingsValue);
                                     return applicationResources.getApplications(bundle.getName())
-                                            .onItem().invoke(applications -> {
-                                                for (Application application : applications) {
-                                                    ApplicationSettingsValue applicationSettingsValue = new ApplicationSettingsValue();
-                                                    applicationSettingsValue.displayName = application.getDisplayName();
-                                                    EmailTemplate applicationEmailTemplate = emailTemplateFactory.get(bundle.getName(), application.getName());
-                                                    settingsValues.bundles.get(bundle.getName()).applications.put(application.getName(), applicationSettingsValue);
-                                                    for (EmailSubscriptionType emailSubscriptionType : EmailSubscriptionType.values()) {
-                                                        if (applicationEmailTemplate.isEmailSubscriptionSupported(emailSubscriptionType)) {
-                                                            settingsValues.bundles.get(bundle.getName()).applications.get(application.getName()).notifications.put(emailSubscriptionType, false);
-                                                        }
-                                                    }
-                                                }
-                                            });
+                                            .onItem().transformToMulti(Multi.createFrom()::iterable)
+                                            .onItem().transformToUniAndMerge(application -> {
+                                                ApplicationSettingsValue applicationSettingsValue = new ApplicationSettingsValue();
+                                                applicationSettingsValue.displayName = application.getDisplayName();
+                                                settingsValues.bundles.get(bundle.getName()).applications.put(application.getName(), applicationSettingsValue);
+                                                return Multi.createFrom().iterable(Arrays.asList(EmailSubscriptionType.values()))
+                                                        .onItem().transformToUniAndMerge(emailSubscriptionType -> {
+                                                            // TODO NOTIF-450 How do we deal with a failure here? What kind of response should be sent to the UI when the engine is down?
```

Comment:
@josejulio WDYT?

Reply:
mmmmh. We can throw an error for now - in the future this info should live in the database and we won't have this problem.
Other option would be to put this check into common code

Reply:
Let's wait for the templates stored in the DB.